server/app/routes/v1/grids_api.rb

```diff
@@ -127,8 +127,16 @@ class GridsApi < Roda
             end
             unless r['services'].nil?
               services = r['services'].split(',').map do |service|
-                @grid.grid_services.find_by(name: service).try(:id)
-              end.delete_if{|s| s.nil?}
+                if service.include?('/')
+                  stack_name, service_name = service.split('/', 2)
+                  stack_id = @grid.stacks.where(name: stack_name).first.try(:id)
+                  if stack_id && service_name
+                    @grid.grid_services.where(stack_id: stack_id, name: service_name).first.try(:id)
+                  end
+                else
+                  @grid.grid_services.find_by(name: service).try(:id)
```

Comment:
I think this is a bug that also needs to be fixed now that we're making the `?service=` stack-aware. The grid service `name` is not unique within a stack, and there can be multiple services with a matching name (in different stacks), so this `@grid.grid_services.find_by(name: ...)` leads to unpredictable/broken behavior:

```
$ kontena stack install test/test.yaml 
 [done] Creating stack test      
 [done] Triggering deployment of stack test     
 [done] Waiting for deployment to start     
 [done] Deploying service redis     
```

With only a `test/redis` service, `--service redis` returns the logs for that stack service:

```
$ kontena grid logs --tail=1 --service redis
2017-08-22T08:43:00.000Z test.redis-1: 1:M 22 Aug 08:43:00.491 * Ready to accept connections
```

Create a new top-level `redis` service:
```
$ kontena service create redis redis
 [done] Creating redis service      
$ kontena service deploy redis
 [done] Deploying service redis      
⊛ Deployed instance development/null/redis-1 to node core-01
```
```
$ kontena service ls
NAME               INSTANCES   STATEFUL   STATE     EXPOSED PORTS
⊝ redis            1 / 1       no         running
⊝ test/redis       1 / 1       no         running
```

The `grid logs` API is broken: you are unable to request the logs for the `redis` service outside of the `test` stack.

```
$ kontena grid logs --tail=1 --service redis
2017-08-22T08:43:00.000Z test.redis-1: 1:M 22 Aug 08:43:00.491 * Ready to accept connections
```

The actual result varies, and `grid logs --service redis` can also return the logs for the `redis` service, depending on the phase of the moon when the services were created.